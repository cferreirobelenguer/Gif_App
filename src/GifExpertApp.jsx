import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

export const GifExpertApp=()=>{
    const [categories,setCategories]=useState([])

    const onAddCategory=(newCategory)=>{
        //Si la categoría existe que no haga nada, y si sí existe que lo inserte en el state
        if(categories.includes(newCategory)){return;}
        setCategories([newCategory,...categories])
        console.log(newCategory)
    }

    return(
        <div className="card_container">
            <div className="card_title">
                <h1>GifExpertApp</h1>
            </div>
            <AddCategory  onNewCategory={event=>onAddCategory(event)}></AddCategory>
            <div className="card_categories">
                
                    {categories.map(i=>(
                            <div key={i} >
                                <GifGrid 
                                    category={i}
                                    idKey={i}>
                                </GifGrid>
                            </div>
                        )
                    )}
            </div>
        </div>
    )
}