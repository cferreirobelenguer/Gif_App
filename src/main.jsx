import React from 'react'
import ReactDOM from 'react-dom/client'
import { GifExpertApp } from './GifExpertApp'
import '../src/assets/scss/style.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifExpertApp></GifExpertApp>
  </React.StrictMode>
)
