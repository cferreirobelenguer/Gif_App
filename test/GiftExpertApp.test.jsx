import { render,screen,fireEvent} from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";



test("probar",()=>{
    const container=render(<GifExpertApp/>)
    expect(container).toMatchSnapshot();
})