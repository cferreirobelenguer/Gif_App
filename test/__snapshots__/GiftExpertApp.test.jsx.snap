// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probar 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card_container"
      >
        <div
          class="card_title"
        >
          <h1>
            GifExpertApp
          </h1>
        </div>
        <div
          class="card_input"
        >
          <div
            class="form_submit"
          >
            <form
              aria-label="form"
            >
              <input
                class="inputStyle"
                placeholder="Introduce tu gift a buscar"
                required=""
                type="text"
                value=""
              />
            </form>
          </div>
          <div
            class="submit_container"
          >
            <button
              class="btn_submit"
            >
              Enviar
            </button>
          </div>
        </div>
        <div
          class="card_categories"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card_container"
    >
      <div
        class="card_title"
      >
        <h1>
          GifExpertApp
        </h1>
      </div>
      <div
        class="card_input"
      >
        <div
          class="form_submit"
        >
          <form
            aria-label="form"
          >
            <input
              class="inputStyle"
              placeholder="Introduce tu gift a buscar"
              required=""
              type="text"
              value=""
            />
          </form>
        </div>
        <div
          class="submit_container"
        >
          <button
            class="btn_submit"
          >
            Enviar
          </button>
        </div>
      </div>
      <div
        class="card_categories"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
